src/plugins/layout/shadowdom.ts

```diff
@@ -69,59 +62,39 @@ export class ShadowDom {
 
     assert(!!parent, "moveShadowNode", "parent is missing");
     assert(!!shadowNode, "moveShadowNode", "shadowNode is missing");
-    if (!(parent && shadowNode)) {
-      return;
-    }
+    if (parent && shadowNode) {
+      if (this.classifyNodes) {
+        if (!this.hasClass(shadowNode, NewNodeClassName)) {
+          this.setClass(shadowNode, MovedNodeClassName);
+        }
+      }
 
-    if (this.classifyNodes) {
-      this.setClass(shadowNode, MovedNodeClassName);
-      if (this.hasClass(parent as IShadowDomNode, NewNodeClassName)) {
-        this.removeClass(shadowNode, TopNewNodeClassName);
+      if (nextSibling) {
+        parent.insertBefore(shadowNode, nextSibling);
       } else {
-        this.setClass(shadowNode, TopNewNodeClassName);
+        parent.appendChild(shadowNode);
       }
     }
-
-    if (nextSibling) {
-      parent.insertBefore(shadowNode, nextSibling);
-    } else {
-      parent.appendChild(shadowNode);
-    }
     return shadowNode;
   }
 
-  public updateShadowNode(index: number, newLayout?: ILayoutState) {
+  public updateShadowNode(index: number) {
     let shadowNode = this.getShadowNode(index);
-
-    assert(!!shadowNode, "updateShadowNode", "shadowNode is missing");
-    if (!shadowNode) {
-      return;
-    }
-
-    if (shadowNode) {
-      if (newLayout) {
-        shadowNode.layout = newLayout;
-        shadowNode.ignore = (newLayout.tag === IgnoreTag);
-      }
-
-      if (this.classifyNodes) {
-        this.setClass(shadowNode, UpdatedNodeClassName);
-      }
+    if (shadowNode && this.classifyNodes) {
+      this.setClass(shadowNode, UpdatedNodeClassName);
```

Comment:
CAn you describe the states that these are encoding, what each means and what the possible transitions between the states are? A comment that did this would be very helpful for understanding the code.